modules/minion/tests/minion/TaskTest.php

```diff
@@ -67,4 +108,37 @@ public function test_convert_class_to_task($expected, $class)
 	{
 		$this->assertSame($expected, Minion_Task::convert_class_to_task($class));
 	}
+	
+	/**
+	 * Provides test data for test_set_domain_name()
+	 *
+	 * @return array
+	 */
+	public function provider_set_domain_name()
+	{
+		return [
+			['https://www.example.com/welcome', 'https://www.example.com', 'welcome'],
+			['http://www.example.com/welcome', 'http://www.example.com', 'welcome'],
+			['http://www.example2.com/welcome', NULL, 'welcome'],
+			['http://www.example.com:8080/welcome', 'http://www.example.com:8080', 'welcome'],
+		];
+	}
+
+	/**
+	 * Tests that a the domain name can be configured for url's
+	 *
+	 * @covers Minion_Task::set_domain_name
+	 * @dataProvider provider_set_domain_name
+	 * @param string Expected domain url
+	 * @param string Input domain name
+	 */
+	public function test_set_domain_name($expected, $name, $uri)
+	{
+		Minion_Task::set_domain_name($name);
+		
+		$this->assertSame(
```

Comment:
Static function should not be called dynamic

Reply:
It's a call to the Minion_Task class, not using $this.

Reply:
github web highlighted the wrong lines....my comment is related to line 139

`$this->assertSame()`

should be replaced with:

`self::assertSame()`

Reply:
Test function is not a static function, neither is `assertSame()`. So I still not get the issue.

Reply:
`Minion_TaskTest`  extends from `KO7_Unittest_TestCase` which extends from `TestCase`, which extends from `Assert` -> where the `assertSame` function is defined.

Here is the defined function:

![Screenshot_3](https://user-images.githubusercontent.com/42447585/54279050-5d0b2a80-4594-11e9-84af-6a06bd6dc029.png)

deinfed **`static`**. I think it works via `$this->assertSame` because of Magic Methods (which should be kinda avoided), but inside the PHPUnit Framework it is also called via:
`static::assertSame`, so i thought we should also use it that way. Anyway, if you still want to use `$this` feel free to merge!

Reply:
Interesting...
If you look at the documentation of php unit the [usage example](https://phpunit.de/manual/6.5/en/appendixes.assertions.html#appendixes.assertions.assertSame) of `assertSame`

[This appendix](https://phpunit.de/manual/6.5/en/appendixes.assertions.html#appendixes.assertions.static-vs-non-static-usage-of-assertion-methods) has some more details on it. It says that there is no right or wrong way.
From my perspective I would agree with you because it's a static function so it should be used as a static function.

**However**: all tests in the Koseven use the `$this` invocation, the documentation of phpunit also the dynamic call, so I would stick with that.

But again: interesting :thinking:

Reply:
> If you look at the documentation of php unit the usage example of assertSame

It's hilarious...sometimes in PHPUnit they use `$this` sometimes they use `static`

> However: all tests in the Koseven use the $this invocation, the documentation of phpunit also the dynamic call, so I would stick with that.

Hm...yes kinda agree...since i need to rewrite many unittests i think we should move to static calls, because magic methods are kinda....yeah....you know 🙈...also my IDE warns my about this every time and it triggers me a bit 😄, but maybe in the future we can vote about that, let's keep it as it as....great work by the way! 😃 

Just last clearification for me: will it be merged to devel only as a future? Or to both devel and master?

Reply:
p.s first paragraph in the new manual for version 8.0 [Link](https://phpunit.readthedocs.io/en/8.0/assertions.html?highlight=assertSame)

> A common question, especially from developers new to PHPUnit, is whether using $this->assertTrue() or self::assertTrue(), for instance, is “the right way” to invoke an assertion. The short answer is: there is no right way. And there is no wrong way, either. It is a matter of personal preference.

😺